### PR TITLE
Add warning when typing in instance name

### LINKED ISSUE #5489: Show warning to users using manual instance name

### PR DESCRIPTION
**Pull request by seadowg**
Closes #5489

A warning will appear when adding/editing an instance name (in the "Save as" field) at the end of a form.

#### What has been done to verify that this works as intended?

New tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

A few choices to discuss here:

* The warning is shown as soon as the user taps on/focuses on the instance name field.
* I opted to use a light grey instead of a light blue for our theme's ['Surface Variant' color](https://m3.material.io/styles/color/the-color-system/key-colors-tones) (which is used as the background for Filled Cards). This was because it seemed more consistent with the Material 3 examples I looked at.
* I moved the "Learn more..." out of the text (where it was as a link) and into a Text Button in the standard space for Card actions. Links in text are far harder to implement than distinct buttons, and I don't think what I've got for has any drawbacks (as far as I'm aware).
* I decided to have the forum post open in the system browser in a new task. This way, the user can multitask or press back to return to Collect. I figured giving them options for returning to form entry while also keeping the relevant information up would be useful if they needed to then send that/show it to someone else.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Just adds the warning, so not a lot of risk here. This can probably merge without QA provided the reviewers have a quick play with it.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by lognaturel — APPROVED** [review]
Works for me! Tests and code look good and I ran it.

If we end up using this component more we can iterate on the look and feel but given how few people will see this I think it's good enough. I probably would have gone with the blue background. I also do prefer the inline link over the button (but again, not important).

File: collect_app/src/main/res/layout/form_entry_end.xml [hunk 1]
```diff
@@ -11,6 +11,8 @@ the specific language governing permissions and limitations under the License.
 <androidx.core.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (grzesiek2010) [inline collect_app/src/main/res/layout/form_entry_end.xml:12]: Usually, we don't mix `android:` with `xmlns:` and have `xmlns:` first then `android:` and `app:` at the end.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 10

**Verified Cases:**
- changing instance name in Fill Blank Form and Edit Saved Form
- checking "Learn more” in the dialog
- form with a generated instance name
- rotating, minimizing, locking the screen
- dark/light mode
- don’t keep activities enabled/disabled

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Android 13

**Pull request merged**